astrobee/survey_manager/scripts/command_astrobee

```diff
@@ -0,0 +1,308 @@
+    if bay == "jem_bay6":
+        return "'11, -9, 4.8'"
+    if bay == "jem_bay7":
+        return "'11, -9.7, 4.8'"
+
+class SurveyExecutor:
+
+    def __init__(self, robot_name, goal, bay, run):
+        self.robot = robot_name
+        self.input_path = '/tmp/input_' + robot_name
+        self.output_path = '/tmp/output_' + robot_name
+
+        # Check if the file exists
+        if os.path.exists(self.input_path):
+            os.remove(self.input_path)
+        if os.path.exists(self.output_path):
+            os.remove(self.output_path)
+
+        # Declare socket for process input
+        self.sock_input = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+        self.sock_input.settimeout(1)  # Set a timeout for socket operations
+        self.sock_input.bind(self.input_path)
+        self.sock_input.listen(1)  # Listen for one connection
+        # self.sock_input_connected = False
+        # self.sock_input.accept(connect_input_callback)
+
+        # Declare socket for process output
+        self.sock_output = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+        self.sock_output.settimeout(1)  # Set a timeout for socket operations
+        self.sock_output.bind(self.output_path)
+        self.sock_output.listen(1)  # Listen for one connection
+        # self.sock_output_connected = False
+        # self.sock_output.accept(connect_output_callback)
+
+        # Open the input FIFO file descriptor with non-blocking flag
+        # self.input_fifo_fd = os.open(self.input_fifo_path, os.O_RDONLY | os.O_NONBLOCK)
+        # self.input_fifo = os.fdopen(self.input_fifo_fd)
+
+        # Open the output FIFO file descriptor with non-blocking flag
+        # self.output_fifo_fd = os.open(self.output_fifo_path, os.O_WRONLY)
+        # self.output_fifo = os.fdopen(self.output_fifo_fd, 'w')
+
+
+        self._stop_event = threading.Event()
+
+
+        self.survey_manager_executor(goal, bay, run)
+
+        self.sock_input.close()
+        self.sock_output.close()
+
+    # def connect_input_callback(sock, addr):
+    #     """Callback function that is called when a connection is made to the socket."""
+    #     print("Input connection made from", addr)
+    #     sock_input_connected = False
+
+    # def connect_output_callback(sock, addr):
+    #     """Callback function that is called when a connection is made to the socket."""
+    #     print("Output connection made from", addr)
+    #     sock_output_connected = False
+
+    def thread_write_output(self, process):
+        print("starting thread_write_output...")
+        # Store commulative output
+        output_total = ""
+        connected = False
+        try:
+            while True:
+                # Get output from process
+                output = process.stdout.readline()
+                if (output == '' and process.poll() is not None) or self._stop_event.is_set():
+                    break
+                if output:
+                    output_total += output
```

Comment:
interesting I found otherwise on the internet from what I searched https://stackoverflow.com/questions/32920991/concatenating-strings-faster-than-appending-to-lists

Reply:
Interesting. I think the article you linked is not very relevant because it's doing something more complicated than just joining strings.

However, I just did a bit of testing locally with Python 3.8 on Ubuntu 20.04, and I think these two approaches have basically exactly the same speed now. I can only guess that the Python compiler has gotten smart enough to (in some sense) optimize the string append approach to the list append approach, which is great!

You can see here the difference used to be very large: https://waymoot.org/home/python_string/ (method 1 vs. method 4)

Anyway, I retract the comment.